include/igraph_datatype.h

```diff
@@ -30,6 +30,41 @@
 
 __BEGIN_DECLS
 
+struct igraph_i_property_cache_t;
+typedef struct igraph_i_property_cache_t igraph_i_property_cache_t;
+
+typedef enum {
+    /* Stores whether the graph has at least one self-loop */
+    IGRAPH_PROP_HAS_LOOP = 0,
+
+    /* Stores whether the graph has at least one multi-edge, taking into account
+     * edge directions in directed graphs. In other words, this property should
+     * be false for a directed graph with edges (a, b) and (b, a). */
+    IGRAPH_PROP_HAS_MULTI,
+
+    /* Stores whether the graph has at least one multi-edge, ignoring edge
+     * directions in directed graphs. In other words, this property should
+     * be true for a directed graph with edges (a, b) and (b, a). */
```

Comment:
It might be better to interpret this precisely as the name would suggest, i.e. make it true when there are reciprocal edges. Examples:

```
a -> b, b -> a

MULTI=false, RECIPROCAL=true
```

```
a -> b, a -> b

MULTI=true RECIPROCAL=false
```

```
a -> b, b -> a, b->a

MULTI=true, RECIPROCAL=true
```

In _undirected_ graphs, `RECIPROCAL` would just be ignored.

I think that there are legitimate reasons to have a public function that checks for reciprocal edges, and it would be nice if the meaning of this property matched. That function, however, is outside of the scope of this PR.

Reply:
Fixed in 3d2c043